internal/proxy/impl.go

```diff
@@ -2523,6 +2583,66 @@ func (node *Proxy) Search(ctx context.Context, request *milvuspb.SearchRequest)
 	return qt.result, nil
 }
 
+func (node *Proxy) getVectorPlaceholderGroupForSearchByPks(ctx context.Context, request *milvuspb.SearchRequest) ([]byte, error) {
+	placeholderGroup := &commonpb.PlaceholderGroup{}
+	err := proto.Unmarshal(request.PlaceholderGroup, placeholderGroup)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(placeholderGroup.Placeholders) != 1 || len(placeholderGroup.Placeholders[0].Values) != 1 {
+		return nil, errors.New("PlaceholderGroup is invalid")
+	}
+	queryExpr := string(placeholderGroup.Placeholders[0].Values[0])
+
+	annsField, err := funcutil.GetAttrByKeyFromRepeatedKV(AnnsFieldKey, request.SearchParams)
+	if err != nil {
+		return nil, err
+	}
+
+	queryRequest := &milvuspb.QueryRequest{
+		Base:                  request.Base,
+		DbName:                request.DbName,
+		CollectionName:        request.CollectionName,
+		Expr:                  queryExpr,
+		OutputFields:          []string{annsField},
+		PartitionNames:        request.PartitionNames,
+		TravelTimestamp:       request.TravelTimestamp,
+		GuaranteeTimestamp:    request.GuaranteeTimestamp,
+		QueryParams:           nil,
+		NotReturnAllMeta:      request.NotReturnAllMeta,
+		ConsistencyLevel:      request.ConsistencyLevel,
+		UseDefaultConsistency: request.UseDefaultConsistency,
+	}
+
+	queryResults, _ := node.Query(ctx, queryRequest)
+
+	err = merr.Error(queryResults.GetStatus())
+	if err != nil {
+		return nil, err
+	}
+
+	if queryResults.FieldsData == nil ||
+		len(queryResults.FieldsData) != 2 ||
+		queryResults.FieldsData[0].FieldName != annsField && queryResults.FieldsData[1].FieldName != annsField {
```

Comment:
these checks could be removed, and looking up the vector field could be done by iterating `queryResults.GetFieldsData()`.